sheets/enrollment_change_api.py

```diff
@@ -0,0 +1,266 @@
+                whether or not it was updated.
+        """
+        form_response_id = int(
+            row_data[self.sheet_metadata.FORM_RESPONSE_ID_COL].strip()
+        )
+        user_input_json = json.dumps(
+            self.sheet_metadata.get_form_input_columns(row_data)
+        )
+        with transaction.atomic():
+            enroll_change_request, created = self.request_model_cls.objects.select_for_update().get_or_create(
+                form_response_id=form_response_id,
+                defaults=dict(raw_data=user_input_json),
+            )
+            raw_data_changed = enroll_change_request.raw_data != user_input_json
+            if raw_data_changed:
+                enroll_change_request.raw_data = user_input_json
+                enroll_change_request.save()
+        return enroll_change_request, created, raw_data_changed
+
+    def update_sheet_from_results(self, row_results):
+        """
+        Helper method that updates the relevant spreadsheet cells based on the row results, logs any
+        necessary messages, and returns a summary of the changes made
+
+        Args:
+            row_results (iterable of RowResult): Objects representing the results of processing rows
+
+        Returns:
+            dict: Row results grouped into a dict with the result types as keys
+        """
+        row_result_dict = group_into_dict(
+            row_results, key_fn=op.attrgetter("result_type")
+        )
+        processed_row_results = row_result_dict.get(ResultType.PROCESSED, [])
+        if processed_row_results:
+            self.update_row_completed_dates(processed_row_results)
+        failed_row_results = row_result_dict.get(ResultType.FAILED, [])
+        if failed_row_results:
+            self.update_row_errors(failed_row_results)
+        out_of_sync_row_results = row_result_dict.get(ResultType.OUT_OF_SYNC, [])
+        if out_of_sync_row_results:
+            log.warning(
+                "Rows found without a completed date, but local records indicate that they were completed: %s",
+                [row_result.row_index for row_result in out_of_sync_row_results],
+            )
+            self.update_row_completed_dates(out_of_sync_row_results)
+        ignored_row_results = row_result_dict.get(ResultType.IGNORED, [])
+        if ignored_row_results:
+            log.warning(
+                "Ignored rows in %s: %s",
+                self.sheet_metadata.sheet_name,
+                [row_result.row_index for row_result in ignored_row_results],
+            )
+        return row_result_dict
+
+    def should_ignore_row(self, row_data, completed_form_response_ids):
+        """
+        Indicates whether or not this row should be skipped entirely before attempting to parse.
+
+        Args:
+            row_data (list of str): The raw data of the given spreadsheet row
+            completed_form_response_ids (iterable of int): Response ids of request objects in the
```

Comment:
Type should be something more specific since we're looking up values in it, not just iterating over it